### PR TITLE
feature/refine-active-states/154338987

### PR DESCRIPTION
**Pull request by rachelmcgrane**
https://www.pivotaltracker.com/story/show/154338987

https://github.com/forever-inc/a-new-hope/pull/1109

kapture 2018-01-16 at 11 20 32

### CODE REVIEW

**Comment by alexisBimatovs** [comment]
jumped in here to finish this story so that active state persist through to relevant product pages

some samps below:

screen shot 2018-01-21 at 4 58 50 pm

screen shot 2018-01-21 at 4 58 46 pm

screen shot 2018-01-21 at 4 58 30 pm

File: app/helpers/forever_style_guide/application_helper.rb [hunk 1]
```diff
@@ -70,8 +70,13 @@ def has_ambassador?
     end
 
     #active state nav
-    def is_active?(controller_name)
-      return true if [controller.controller_name, controller.action_name].include? controller_name
+    def is_active?(controller_names, product_type)
```
Comment (katywatkins) [inline app/helpers/forever_style_guide/application_helper.rb]: Definitely not sure this is the best refactor we can do, but food for thought: 
If we remove the arrays and deal product type in the markup, pretty sure we can simplify this down to
```
def is_active?(controller_name, product_type = nil)
      controller.controller_name.include?(controller_name) || controller.action_name.include?(controller_name) || (@product != nil && @product.product_type == product_type && !@product.name.include?('Historian'))
end
```

Which is still awkward, but feels a little bit simpler?

Comment (katywatkins) [reply app/helpers/forever_style_guide/application_helper.rb]: I'm also perplexed by using  'controller_name' when it's not really controller name, but I also haven't come up with a better idea.

**Comment by alexisBimatovs** [comment]
@katywatkins back 2 u

**Comment by katywatkins** [comment]
As discussed in person, we should test this once deployed to ensure the active states display as expected when purchasing printed projects (particularly from Jupiter). Otherwise, looks good! 👍

**Pull request merged**